metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceImpl.java

```diff
@@ -130,6 +130,11 @@ public void notifyOfPartitionAddition(final MetacatSaveTablePartitionPostEvent e
             tableMessage, event.getName(),
             "Unable to publish table partition add notification",
             Metrics.CounterSNSNotificationTablePartitionAdd.getMetricName());
+        //publish the delete column key metric after publishing message
+        if (partitionsUpdatePayload.getLatestDeleteColumnValue() != null) {
```

Comment:
Are we publishing atlas metric in all cases?

Reply:
only when it has the keys attached.

Reply:
we should be in all cases to know when the keys are not attached?